dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_BasicsTests/AutoScaleMethodsTests.cs

```diff
@@ -0,0 +1,106 @@
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, _GroupName);
+            Assert.IsTrue(success, "Could not delete the group.");
+        }
+
+        [TestMethod()]
+        public async Task DeleteAutoScalingGroupGroupNonexistentTest()
+        {
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, "NonExistentSite");
+            Assert.IsFalse(success, "Could not find the group to delete.");
+        }
+
+        [TestMethod()]
+        public async Task TerminateInstanceInAutoScalingGroupTest()
+        {
+            var instanceId = string.Empty;
+            var success = await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(_Client, instanceId);
+            Assert.IsTrue(success, "Could not terminate the instance.");
+        }
+
+        [TestMethod()]
+        public async Task DescribeScalingActivitiesTest()
+        {
+            var activities = await AutoScaleMethods.DescribeAutoScalingActivitiesAsync(_Client, _GroupName);
+            Assert.IsTrue(activities.Count > 0, "Can't find any auto scaling activities for the group.");
+        }
+
+        [TestMethod()]
+        public async Task SetDesiredCapacityTest()
+        {
+            var success = await AutoScaleMethods.SetDesiredCapacityAsync(_Client, _GroupName, 3);
+            Assert.IsTrue(success, "Couldn't set the desired capacity.");
+            var groups = await AutoScaleMethods.DescribeAutoScalingGroupsAsync(_Client, _GroupName);
+            groups.ForEach(group =>
+            {
+                if (group.AutoScalingGroupName == _GroupName)
+                {
+                    Assert.AreEqual(3, group.DesiredCapacity, "Capacity was not set properly.");
+                }
+            });
+        }
+
+        [TestMethod()]
+        public async Task DescribeAccountLimitsTest()
+        {
+            var success = await AutoScaleMethods.DescribeAccountLimitsAsync(_Client);
+            Assert.IsTrue(success, "Couldn't retrieve account limits.");
+        }
+
+        [TestMethod()]
+        public async Task DescribeAutoScalingGroupsTest()
+        {
+            var details = await AutoScaleMethods.DescribeAutoScalingGroupsAsync(_Client, _GroupName);
+            Assert.IsTrue(details.Count > 0, "Couldn't find that Auto Scaling group.");
+        }
+
+        [TestMethod()]
+        public async Task EnableMetricsCollectionTest()
+        {
+            var success = await AutoScaleMethods.EnableMetricsCollectionAsync(_Client, _GroupName);
+            Assert.IsTrue(success, $"Couldn't enable metrics collection for {_GroupName}.");
+        }
+
+        [TestMethod()]
+        public async Task UpdateAutoScalingGroupTest()
+        {
+            var success = await AutoScaleMethods.UpdateAutoScalingGroupAsync(_Client, _GroupName, _LaunchTemplateName, _ServiceLinkedRoleArn, 3);
+            Assert.IsTrue(success, "Couldn't update the Auto Scaling group: {_GroupName}.");
```

Comment:
Should this be an interpolated string like `EnableMetricsCollectionTest` above? ($)

Reply:
Yes. Fixed!